packages/network/test/unit/agent_spec.ts

```diff
@@ -491,6 +513,136 @@ describe('lib/agent', function () {
     })
   })
 
+  context('CombinedAgent with CA overrides', function () {
+    const proxyUrl = `https://localhost:${PROXY_PORT}`
+
+    before(function () {
+      this.servers = new Servers()
+
+      return this.servers.start(HTTP_PORT, HTTPS_PORT)
+    })
+
+    after(function () {
+      return this.servers.stop()
+    })
+
+    ;[
+      {
+        name: 'should use the npm_config_cafile override',
+        npm_config_cafile: path.join(tempDirPath, 'cafile.pem'),
+        npm_config_ca: 'ca',
+        NODE_EXTRA_CA_CERTS: path.join(tempDirPath, 'extra_ca_certs.pem'),
+        caContents: [caFileContents],
+        option: 'npm_config_cafile',
+      },
+      {
+        name: 'should use the npm_config_ca override',
+        npm_config_ca: caContents,
+        NODE_EXTRA_CA_CERTS: path.join(tempDirPath, 'extra_ca_certs.pem'),
+        caContents: [caContents],
+        option: 'npm_config_ca',
+      },
+      {
+        name: 'should use the NODE_EXTRA_CA_CERTS override',
+        NODE_EXTRA_CA_CERTS: path.join(tempDirPath, 'extra_ca_certs.pem'),
+        caContents: [extraCACertsFileContents, ...tls.rootCertificates],
+        option: 'NODE_EXTRA_CA_CERTS',
+      },
+    ].slice().map((testCase) => {
+      context(testCase.name, function () {
+        beforeEach(function () {
+          // PROXY vars should override npm_config vars, so set them to cause failures if they are used
```

Comment:
Feels like these tests should set NODE_TLS_REJECT_UNAUTHORIZED = '1' in beforeEach, since there should be no TLS errors.

Reply:
Great idea. Took a little bit of time to get all the certificates talking to each other properly, but this is now working.